### PR TITLE
Avoid failing on shutdown executor

### PR DESCRIPTION
**Pull request by yschimke**
Testing of https://github.com/square/okhttp/issues/7410

Consider this fix for discussion.  Open to suggestions.

Issue happens because RealCall.executeOn has rejected execute call and falls back to recursive handling.

### CODE REVIEW

**Comment by yschimke** [comment]
Pretty sure this is not the right fix, since Executor could be shutdown at any point.  So will need to change AsyncCall.executeOn.

**Comment by yschimke** [comment]
Since promoteAndExecute is in the recursive path, this should work.  But still open for suggestions here.

File: okhttp/src/jvmMain/kotlin/okhttp3/Dispatcher.kt [hunk 1]
```diff
@@ -15,9 +15,7 @@
  */
 package okhttp3
 
-import java.util.ArrayDeque
-import java.util.Collections
-import java.util.Deque
+import java.util.*
```
Comment (JakeWharton) [inline okhttp/src/jvmMain/kotlin/okhttp3/Dispatcher.kt:18]: Revert star import. Seems like we need kotlinter/ktlint setup in here!

File: okhttp/src/jvmMain/kotlin/okhttp3/internal/connection/RealCall.kt [hunk 2]
```diff
@@ -27,16 +27,7 @@ import java.util.concurrent.atomic.AtomicBoolean
...
-import okhttp3.Call
-import okhttp3.Callback
-import okhttp3.CertificatePinner
-import okhttp3.EventListener
-import okhttp3.HttpUrl
-import okhttp3.Interceptor
-import okhttp3.OkHttpClient
-import okhttp3.Request
-import okhttp3.Response
+import okhttp3.*
```
Comment (JakeWharton) [inline okhttp/src/jvmMain/kotlin/okhttp3/internal/connection/RealCall.kt:30]: Revert

File: okhttp/src/jvmTest/java/okhttp3/DispatcherCleanupTest.kt [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  fun testFinish() {
+    val okhttp = OkHttpClient()
+    val callback = object : Callback {
+      override fun onFailure(call: Call, e: IOException) {}
+      override fun onResponse(call: Call, response: Response) {
+        response.close()
+      }
+    }
+    repeat(10_000) {
+      okhttp.newCall(Request.Builder().url("https://example.com").build()).enqueue(callback)
```
Comment (JakeWharton) [inline okhttp/src/jvmTest/java/okhttp3/DispatcherCleanupTest.kt:32]: Probably needs MockWebServer so as not to be dependent on the internet.

**Review by JakeWharton — APPROVED** [review]
Just a slight approve as I'm out my depth here.

File: okhttp/src/jvmMain/kotlin/okhttp3/Dispatcher.kt (hunk 1 above)
Comment (yschimke) [reply okhttp/src/jvmMain/kotlin/okhttp3/Dispatcher.kt:18]: Sorry, new laptop.

**Comment by yschimke** [comment]
@swankjesse happy to revisit after your review. In the mean time, give it some time to bake.  Should only have any effect, after Executor is shutdown, so not likely to cause problems in normal use.

**Pull request merged**